active/001-tabular-msa.md

```diff
@@ -0,0 +1,292 @@
+- `sort(key=None, reverse=False) -> None`
+
+   Sort sequences in-place. If `key=None` the existing keys will be used to
+   sort. Otherwise `key` matches the constructor.
+- `merge(UNRESOLVED) -> TabularMSA`
+
+   Merge two `TabularMSA` objects.
+- `consensus() -> Sequence`
+
+   Return majority consensus sequence matching alignment's `dtype`. Other
+   types of consensus may be added later.
+- `gap_frequencies(axis, relative=False) -> np.ndarray`
+
+   Return vector of gap frequencies along an axis.
+- `entropies(axis) -> np.ndarray`
+
+   Return vector of Shannon entropies along an axis.
+- `reindex(key=None) -> None`
+
+   Assign new keys to sequences in the alignment. `key` matches constructor.
+
+# Drawbacks
+
+- scikit-bio users will have to update their code to use a new API.
+
+- The new class name `TabularMSA` may not be as obvious as to its behavior as
+the current class name `Alignment`.
+
+- It is unclear if it is advisable to use a tabular (row-column) MSA as it
+forces an ordering where none exists. See
+[Lee et al. 2002](http://bioinformatics.oxfordjournals.org/content/18/3/452.short).
+
+# Alternatives
+
+We considered representing an MSA internally as a DAG in support of
+partial-order MSAs. However, representing positions in such an MSA defeats the
+entire purpose of using a DAG at all.
+
+We considered *only* supporting partial-order MSAs. However, since scikit-bio
+is a bioinformatics data munging library, we need to support row-column MSAs
+because most bioinformatics tools use or produce these structures.
+
+We considered the following alternative names for the new class:
+
+- `LinearMSA`: it is not immediately obvious that this is a row-column
+alignment
+- `TotalOrderMSA`: even less obvious and it sounds "better" than partial-order
+if users aren't clear on the definitions
+- `RowColMSA` / `RowColumnMSA`: painful to type and hard to remember (was it
+"col" or "column"?)
+- `StandardAlignment`: implies "best" or preferred alignment structure
+- `Alignment`: too general as it doesn't describe an MSA, also implies a
+default alignment structure
+
+# Unresolved questions
+
+Is it useful to merge two `TabularMSA` objects? What does this interface look
+like? Can both axes be merged? If so, do we support the standard SQL join
+operations? Should it be called `join` instead of `merge`?
+
+We are almost matching the `dict` interface for `keys` and `items`; however
+we do not return views, only an iterator. Should these methods be
+renamed/become properties or is this discrepancy a non-issue?
+
+How do we want to deal with axes for methods? Currently the only two methods
+which would need an axis are `entropies` and `gap_frequencies`. Should an
+`axis` parameter be required? If not, the more natural default for these
+methods would be positions, but this does not match the primary axis
+(sequences). Additionally are axes keywords plural ("sequences", "positions")
+or singular ("sequence", "position")?
```

Comment:
I like using the singular form (`axis='sequence'`, `axis='position'`) because it is slightly shorter to type and may be easier to remember than a mixture of singular and plural. This is also consistent with numpy-style axes (`axis=0`, `axis=1`), which do not make a statement about dependence of axes. As a user I get tripped up by pandas' `'index'` and `'columns'`.